src/main/java/com/minecolonies/coremod/colony/buildings/moduleviews/CraftingModuleView.java

```diff
@@ -44,34 +95,63 @@ public JobEntry getJobEntry()
         return this.jobEntry;
     }
 
+    /**
+     * Check if recipes can be taught.
+     * @return true if so.
+     */
+    public boolean isRecipeAlterationAllowed()
+    {
+        return canLearnCraftingRecipes || canLearnFurnaceRecipes;
+    }
+
     /** True if this module can be taught crafting recipes. */
     public boolean canLearnCraftingRecipes() { return this.canLearnCraftingRecipes; }
     /** True if this module can be taught smelting recipes. */
     public boolean canLearnFurnaceRecipes() { return this.canLearnFurnaceRecipes; }
     /** True if this module can be taught 3x3 crafting recipes. */
     public boolean canLearnLargeRecipes() { return this.canLearnLargeRecipes; }
 
+    /**
+     * Unique id of the crafting module view.
+     * @return the id.
+     */
+    public String getId()
+    {
+        return this.id;
+    }
+
     @Override
     public boolean isPageVisible()
     {
-        return false;   // todo this later
+        return isVisible;
     }
 
     @Override
+    @OnlyIn(Dist.CLIENT)
     public Window getWindow()
     {
-        return null;    // todo this later
+        return new WindowListRecipes(buildingView, Constants.MOD_ID + ":gui/layouthuts/layoutlistrecipes.xml", this);
     }
 
     @Override
     public String getIcon()
     {
-        return "crafting";
+        return id;
     }
 
     @Override
     public String getDesc()
     {
         return "com.minecolonies.coremod.gui.workerhuts.recipe";

```

Comment:
This might want to be different based on id as well, so you can have a "Crafting" tab and a "Smelting" tab.  (Though I'm still dubious that "custom" should be separate from "crafting", especially in the UI.)